File: is_anagram/is_anagram.py
# Methods return True if the strings passed are anagram of each other.
# Note: Args expected are str ONLY

def anagram(s1,s2):    

    if len(s1) != len(s2):
        return False
    
    s1 = s1.lower()
    s2 = s2.lower()
    
    for i in s1:
        for j in s2:
            if i not in s2 or j not in s1:
                return False
        
    return True


def anagram_1(s1,s2):

    a = [ord(i) for i in s1]
    b = [ord(i) for i in s2]

    if sum(a) != sum(b):
        return False
    return True
    

def anagram_2(s1,s2):
    
    s1 = list(s1)
    s2 = list(s2)
    s1.sort()
    s2.sort()    
    if s1 == s2:
        return True
    return  False

File: is_anagram/is_anagram.cpp
#include <iostream>
#include <vector>
using namespace std;

bool is_anagram(string str1, string str2) {

	vector<int> count1(256, 0), count2(256, 0);

	if (str1.length() != str2.length())
		return false;

	for (int i = 0; i < str1.length(); i++) {
		++count1[str1[i]];
		++count2[str2[i]];
	}

	for (int i = 0; i < 256; i++)
		if (count1[i] != count2[i])
			return false;

	return true;
}

int main() {

	bool anagram = is_anagram("LISTEN", "SILENT");

	if (anagram)
		cout << "The strings are Anagram of each other" << endl;
	else
		cout << "The strings are not Anagram of each other" << endl;

	return 0;
}
